[tests] Workload tests follow up

radical opened the pull request:
- Remove unused workload install target
- BuildEnvironment: always clean the local nuget cache on CI
- Move all the RequiresDocker attribute classes to the Common directory
- Improve failure message for workload tests when sdk+workload is not installed
- Handle IOExceptions when deleting `testroot` to avoid failing the tests if something is in use by a running process
- Install playwright dependencies on windows

###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4260)

radical committed:
Remove unused workload install target

radical committed:
[tests] BuildEnvironment: always clean the local nuget cache on CI

radical committed:
Move all the RequiresDocker attribute classes to the Common directory

radical committed:
Improve failure message for workload tests when sdk+workload is not installed

radical committed:
cleanup

radical committed:
Handle ioexceptions when deleting testroot

radical committed:
always install playwright+chromium on windows

eerhardt commented on tests/Aspire.Components.Common.Tests/RequiresDockerAttribute.cs:
```diff
@@ -1,12 +1,13 @@
+namespace Aspire.Components.Common.Tests;
 
-[TraitDiscoverer("Aspire.Workload.Tests.RequiresDockerDiscoverer", "Aspire.Workload.Tests")]
+// class level attribute
```
What is this comment for?

radical replied:
`RequiresDockerAttribute` - the name by itself seemed unclear on what it was meant for. Maybe I should make this applicable to methods also.

eerhardt replied:
It says it applies to Class and Method in the AttributeUsage.

Can we just use this method instead of `RequiresDockerFact` and `RequiresDockerTheory`? Would you just annotate the method as:
```C#
[Fact]
[RequiresDocker]
```

radical replied:
Yeah, that would work.

eerhardt replied:
Don't need to fix it here, but that would be nice to consolidate on a single way to do this.

eerhardt commented on tests/Shared/WorkloadTesting/BuildEnvironment.cs:
```diff
@@ -166,19 +168,66 @@ public BuildEnvironment(bool useSystemDotNet = false, TestTargetFramework target
+            {
+                if (EnvironmentVariables.IsRunningOnCI)
+                {
+                    // Investigate failures on CI, so don't guard against exceptions
```
What does this comment mean?

radical replied:
Basically that we want to fail on CI if deleting the `testroot` throws an exception, as it would be something that should be checked manually, like some lingering process from a previous run holding onto a file handle in `testroot`.

eerhardt replied:
>  that we want to fail on CI if deleting the testroot throws an exception

How does swallowing the exception cause the test to fail?

radical replied:
OOPS!!

eerhardt commented on tests/Shared/WorkloadTesting/BuildEnvironment.cs:
```diff
@@ -166,19 +168,66 @@ public BuildEnvironment(bool useSystemDotNet = false, TestTargetFramework target
-        if (Directory.Exists(TestRootPath))
+        Console.WriteLine($"*** [{TargetFramework}] Using path for projects: {TestRootPath}");
+        CleanupTestRootPath();
+        if (!Directory.Exists(TestRootPath))
```
You shouldn't need to check before. You can just call `Directory.CreateDirectory`.

radical committed:
address review feedback from @ eerhardt

radical committed:
Merge remote-tracking branch 'origin/main' into workload-tests-misc-fixes

radical committed:
address review feedback from @ eerhardt

radical committed:
Remove RequiresDockerTheory/RequiresDockerFact attributes

.. in favor of one `RequiresDocker` attribute.

Changes suggested by @ eerhardt .

radical committed:
Fix RequiresDockerDiscoverer's assembly name

radical committed:
Use ProjectReference to Aspire.Components.Common.Tests instead of directly importing RequiresDocker*.cs files, so the xunit discoverer can run

radical committed:
Merge branch 'dotnet:main' into workload-tests-misc-fixes
